### PR TITLE
#395  fix, short options do not understand adjacent values

### PR DESCRIPTION
**Pull request by TheRoSS**
Example of the bug:

commander.option("-c <value>")
console.log("C:", commander.C)

$ node program -c234
C: -2

### CODE REVIEW

**Comment by shadowspawn** [comment]
It looks like this pull request works with `-a234 => a=234` for an option taking a value, and `-xyz => -x -y -z`  if x is a boolean flag, but leaves some combination cases like `-axz 3` or `-xa234`. 

It would be nice if there was a standard we could refer to for short flag handling. Should we only consider numbers?

Keeping pull request open for now.

**Comment by shadowspawn** [comment]
Link to issue: #395

**Comment by shadowspawn** [comment]
Found a couple of posix references which align with suggested pull request:

1) https://www.gnu.org/software/libc/manual/html_node/Argument-Syntax.html

> Multiple options may follow a hyphen delimiter in a single token if the options do not take arguments. Thus, ‘-abc’ is equivalent to ‘-a -b -c’.

2) https://www.gnu.org/software/libc/manual/html_node/Example-of-Getopt.html#Example-of-Getopt

```bash
% testopt -a -b
aflag = 1, bflag = 1, cvalue = (null)

% testopt -ab
aflag = 1, bflag = 1, cvalue = (null)

% testopt -c foo
aflag = 0, bflag = 0, cvalue = foo

% testopt -cfoo
aflag = 0, bflag = 0, cvalue = foo
```

**Comment by TheRoSS** [comment]
Good day. I am rather busy just now. But can participate in this issue in the end of May

File: index.js [hunk 1]
```diff
@@ -579,7 +580,9 @@ Command.prototype.normalize = function(args) {
   var ret = []
     , arg
     , lastOpt
-    , index;
+    , index
+    ,short
```
Comment (shadowspawn) [inline index.js]:
```suggestion
    , short
```

File: index.js [hunk 2]
```diff
@@ -579,7 +580,9 @@ Command.prototype.normalize = function(args) {
...
+    ,opt;
```
Comment (shadowspawn) [inline index.js]:
```suggestion
    , opt;
```

**Comment by shadowspawn** [comment]
Merging into `release/v3.0.0` branch

**Comment by shadowspawn** [comment]
(I deleted some requested style changes that turned out to be stale, format on master had changed.)

**Comment by shadowspawn** [comment]
I made some minor changes, and already had one approval. Merged into v3 and will be included in that release. Closing to make it clear that should not be merged into master.

**Comment by shadowspawn** [comment]
Update: I am working my way through the parsing process looking into another bug. I suspect this fix only works for action based commands.

**Comment by shadowspawn** [comment]
Available now as a prerelease. See #1001

**Comment by shadowspawn** [comment]
Shipped in v3: https://github.com/tj/commander.js/releases/tag/v3.0.0

**Pull request closed without merging**